### PR TITLE
Flow control checkbox widget

### LINKED ISSUE #355: Create the Flow Control Checkbox widget
### Feature description 

As part of the flow control for each course, a user should not be able to proceed to the next lesson, until the current unit is complete.

In order for a user to acknowledge that they've completed the current unit, they should (at minimum) check a box that indicates they've completed the unit.

This issue is to create a checkbox widget that we can use for the flow control of course pages.

### Additional information

[Figma design board](https://www.figma.com/file/KCfjq5xQHg0C1JmcrE3CGx/1332-Private-Experience?node-id=1881%3A160652&t=BhvXTtvCuiYb9AR9-4)

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes

Resolves #355 

## Implementation details

- Adds a Checkbox widget
- If the box is checked, an event is emitted 
- Added a Goal to the Checkbox
- Updated `widgets/README.md` w/ latest emits info

## How to use

```md
:: Checkbox(goal="unit-complete" progress="1")
::
```

**Note** feel free to use this Jupyter Notebook w/ the Checkbox for local testing: 
[checkbox.ipynb.zip](https://github.com/Qiskit/saiba/files/10127415/checkbox.ipynb.zip)

## Screenshots

https://user-images.githubusercontent.com/6276074/205375724-39fb1e1e-9671-4780-9951-e1394ac99c8d.mov

### CODE REVIEW

**Review by abdonrd — COMMENTED** [review]
Looks good, but probably @korgan00 can make a better review!

**Review by yoshiemuranaka — COMMENTED** [review]
Looks like the event name maybe needs to be updated for the progressions to work, maybe we can also update the [readme](https://github.com/Qiskit/saiba/blob/main/widgets/README.md)?

Also maybe a question for @korgan00 does the `goal` prop need to follow a particular casing?  (saw most of the quiz goals were kebab case)

Screenshot 2022-12-02 at 10 05 31 AM

File: widgets/components/checkbox/index.vue [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  <div>
+    <bx-checkbox
+      label-text="I have completed this lesson."
+      @bx-checkbox-changed="handleClick"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+let checked = false;
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply widgets/components/checkbox/index.vue]: removed w/ this commit https://github.com/Qiskit/saiba/pull/358/commits/f5905e3048f34f9b4a79a7d27d8b80ac5a6cd5f3

**Review by yoshiemuranaka — APPROVED** [review]
🎉

**Pull request merged**